tests/test_psygnal.py

```diff
@@ -988,3 +988,53 @@ def callback() -> None:
 
     with pytest.raises(RecursionError):
         s.emit()
+
+
+def test_signal_order():
+    """Test that signals are emitted in the order they were connected."""
+    emitter = Emitter()
+    mock1 = Mock()
+    mock2 = Mock()
+
+    def callback(x):
+        if x != 10:
+            emitter.one_int.emit(10)
+
+    emitter.one_int.connect(mock1)
+    emitter.one_int.connect(callback)
+    emitter.one_int.connect(mock2)
+    emitter.one_int.emit(1)
+
+    mock1.assert_has_calls([call(1), call(10)])
+    mock2.assert_has_calls([call(1), call(10)])
+
+
+def test_signal_order_suspend():
+    """Test that signals are emitted in the order they were connected."""
+    emitter = Emitter()
+    mock1 = Mock()
+    mock2 = Mock()
+
+    def callback(x):
+        if x < 10:
+            emitter.one_int.emit(10)
+
+    def callback2(x):
+        if x == 10:
+            emitter.one_int.emit(11)
+
+    def callback3(x):
```

Comment:
perhaps callback3 here is intended to simulate what the evented set would do? (without having to use evented set) ... but you could also add the following test for evented set just for the sake of thoroughness:

```python
def test_set_emission_order():
    s = EventedSet()

    def callback1():
        if 1 not in s:
            s.add(1)

    def callback2():
        if 5 not in s:
            s.update(range(5, 10))

    s.events.items_changed.connect(callback1)
    s.events.items_changed.connect(callback2)
    mock = Mock()
    s.events.items_changed.connect(mock)

    s.add(11)
    mock.assert_has_calls(
        [
            call((11,), ()),
            call((1,), ()),
            call((5, 6, 7, 8, 9), ()),
        ]
    )
```